src/core/unittest/SettingsTest.cpp

```diff
@@ -334,6 +336,277 @@ TEST(SettingsTest, StreamRecvWindowDefaultGetsOverridenByIndividualLimits)
 
 #define SETTINGS_SIZE_THRU_FIELD(SettingsType, Field) \
     (FIELD_OFFSET(SettingsType, Field) + sizeof(((SettingsType*)0)->Field))
+TEST(SettingsTest, QuicSettingsSetDefault_SetsAllDefaultsWhenUnset)
+{
+    QUIC_SETTINGS_INTERNAL Settings;
+    CxPlatZeroMemory(&Settings, sizeof(Settings));
+
+    // Set all IsSet fields to 0 to simulate unset state
+    Settings.IsSetFlags = 0;
+
+    QuicSettingsSetDefault(&Settings);
+
+    // Spot-check a few representative fields (add more as needed)
+    ASSERT_EQ(Settings.SendBufferingEnabled, QUIC_DEFAULT_SEND_BUFFERING_ENABLE);
```

Comment:
Question: Does this test bring real value?

It does look to me like it repeats exactly what the code would write but is the type of test that one need to modify with any modification of the code under test (so it will catch no regression, a test failure only means the test needs to be updated).

Reply:
It does feel like a test that check that if the code does `a = b`, then `a == b`: it is only redundancy.
IMO, this tend to bring a low value (if I add a new setting, it is likely that if I get the code wrong, I get the test wrong), and to be a maintenance burden (it needs to be updated with every update to the setting struct).

I'd personally remove it, but that's nitpicking, so up to you.